app/models/book_csv_importer.rb

```diff
@@ -0,0 +1,62 @@
+class BookCsvImporter
+  def initialize(csv_file_content)
+    @csv_file_content = csv_file_content
+  end
+
+  def import!
+    csv_table = CSV.parse(@csv_file_content, headers: true)
+
+    if csv_table.empty?
+      raise "The file is empty"
+    end
+
+    if csv_table.headers.join(",") != "Name,ISBN,Authors,Genres,Featured,Published At,"
+      raise "The headers are incorrect"
```

Comment:
Typically I like to make my error messages as informative as possible.

In this case, I think it would be valuable for the error message to also let the caller know what the headers should be.

That way when someone sees the error message they also know what to do about it.

You could even go the extra mile, something like:

"The headers are incorrect. You are missing A, B, and C"

rather than listing all the headers.

Reply:
On a related note - making informative error messages can be good, but it's not something you should typically invest huge amounts of time on unless it's really important.

If you spend half your implementation time on a feature writing really informative messages, it may not have been worth the effort.

Like almost everything else, it will need a case-by-case decision as to how far you go.